pkg/registry/discovery/endpointslice/strategy.go

```diff
@@ -96,3 +102,38 @@ func (endpointSliceStrategy) ValidateUpdate(ctx context.Context, new, old runtim
 func (endpointSliceStrategy) AllowUnconditionalUpdate() bool {
 	return true
 }
+
+// dropTerminatingConditionOnCreate will drop the terminating condition if the
+// EndpointSliceTerminatingCondition is disabled. Otherwise the field is left untouched.
+func dropTerminatingConditionOnCreate(endpointSlice *discovery.EndpointSlice) {
+	if utilfeature.DefaultFeatureGate.Enabled(features.EndpointSliceTerminatingCondition) {
+		return
+	}
+
+	for i := range endpointSlice.Endpoints {
+		endpointSlice.Endpoints[i].Conditions.Terminating = nil
+	}
+}
+
+// dropTerminatingConditionOnUpdate will drop the terminating condition field if the EndpointSliceTerminatingCondition
+// feature gate is disabled unless an existing EndpointSlice object has the field already set. This ensures
+// the field is not dropped on rollback.
+func dropTerminatingConditionOnUpdate(oldEPS, newEPS *discovery.EndpointSlice) {
+	if utilfeature.DefaultFeatureGate.Enabled(features.EndpointSliceTerminatingCondition) {
+		return
+	}
+
+	unsetTerminatingCondition := true
```

Comment:
I feel like we discussed this before.  This is not exactly correct - if any old EP has it set, any new EP can use it.  Given the context, I think this is acceptable, but worth commenting.

Reply:
> I feel like we discussed this before. 

Yeah, admittedly this always confuses me though :P 

Added more comments to make this clearer. `dropTerminatingConditionOnUpdate` has some extra comments which should help too.